src/simsopt/mhd/vmec.py

```diff
@@ -322,19 +458,54 @@ def set_dofs(self, x):
     def recompute_bell(self, parent=None):
         self.need_to_run_code = True
 
-    def run(self):
+    def set_profile(self, longname, shortname, letter):
         """
-        Run VMEC, if ``need_to_run_code`` is ``True``.
+        This function is used to set the pressure, current, and/or iota
+        profiles.
         """
-        if not self.need_to_run_code:
-            logger.info("run() called but no need to re-run VMEC.")
+        profile = self.__getattribute__(longname + "_profile")
+        if profile is None:
             return
 
-        if not self.runnable:
-            raise RuntimeError('Cannot run a Vmec object that was initialized from a wout file.')
+        n = self.__getattribute__("n_" + longname)
+        vmec_profile_type = self.indata.__getattribute__("p" + shortname + "_type")
+        if vmec_profile_type[:12] == b'power_series':
+            # Evaluate the new Profile on a Gauss-Legendre grid in s,
+            # so the polynomial fit is well conditioned.
+            nodes, weights = np.polynomial.legendre.leggauss(n)
+            x = nodes * 0.5 + 0.5  # So x is in (0, 1)
+            y = profile(x)
+            poly = np.polynomial.polynomial.Polynomial.fit(x, y, n - 1, domain=[0, 1]).convert().coef
+            logger.debug('Setting vmec ' + longname + f' profile using power series.  x: {x}  y: {y}  poly: {poly}')
+            ax = self.indata.__getattribute__("a" + letter)
+            ax[:] = 0.0
+            ax[:n] = poly
+
+        elif vmec_profile_type[:12] == b'cubic_spline':
+            x = np.linspace(0, 1, n)
+            y = profile(x)
+            logger.debug('Setting vmec ' + longname + f' profile using splines. x: {x}  y: {y}')
+            aux_s = self.indata.__getattribute__("a" + letter + "_aux_s")
+            aux_f = self.indata.__getattribute__("a" + letter + "_aux_f")
+            aux_s[:] = 0.0
+            aux_f[:] = 0.0
+            aux_s[:n] = x
+            aux_f[:n] = y
 
```

Comment:
There are several other profile types allowed in vmec that are splines but not cubic splines, e.g. line_segment and akima_spline. You may be able to support these types without much modification to your code.